desktop/core/src/desktop/lib/fs/ozone/upload.py

```diff
@@ -202,6 +203,12 @@ def __init__(self, request):
 
   def new_file(self, field_name, file_name, *args, **kwargs):
     if self._is_ofs_upload():
+      _, file_type = os.path.splitext(file_name)
+      if RESTRICT_FILE_EXTENSIONS.get() and file_type.lower() in [ext.lower() for ext in RESTRICT_FILE_EXTENSIONS.get()]:
```

Comment:
nit: could be a utility function as this is repeated at 5 places. could be useful later as well.

`if restricted_files():`

Reply:
This is handled cleanly in new upload APIs. Old ones are deprecated.